research/conditional/utils/model_utils.py

```diff
@@ -535,13 +538,21 @@ def get_classes_from_module_names(
             classes.append(ExpertGating)
         elif name == "Softmax":
             classes.append(torch.nn.Softmax)
+        elif name == "TokenChoiceRouter":
+            classes.append(TokenChoiceRouter)
         else:
             raise ValueError(f"Unknown name {name}")
     return tuple(classes)
 
 
 def get_mixed_precision_ignored_classes(args) -> list[Type[torch.nn.Module]]:
-    ignored_classes = [ExpertGating, LayerNorm, _BatchNorm, torch.nn.Softmax]
+    ignored_classes = [
+        ExpertGating,
+        LayerNorm,
+        _BatchNorm,
+        torch.nn.Softmax,
```

Comment:
I def think this should be reverted (see above also)

Reply:
Aaa nie spushowałem